src/pages/CardCreate/Modal/index.tsx

```diff
@@ -0,0 +1,105 @@
+import Styled from './index.style'
+interface CardCreateModalProps {
+  isOpen: boolean
+  setIsOpen: React.Dispatch<React.SetStateAction<boolean>>
+  setCardType: React.Dispatch<React.SetStateAction<CardType | undefined>>
+}
+
+interface TouchableCardColorPickerProps {
+  cardType: CardType
+  setCardType: React.Dispatch<React.SetStateAction<CardType | undefined>>
+  onClicked: () => void
+}
+
+type TouchableCardColorPickerAreEqualsType = (
+  prevProps: Readonly<React.PropsWithChildren<TouchableCardColorPickerProps>>,
+  nextProps: Readonly<React.PropsWithChildren<TouchableCardColorPickerProps>>
+) => boolean
+
+const TouchableCardColorPickerAreEquals: TouchableCardColorPickerAreEqualsType =
+  (prev, curr) => prev.cardType === curr.cardType
+const TouchableCardColorPicker = React.memo(
+  ({ cardType, setCardType, onClicked }: TouchableCardColorPickerProps) => {
+    const onClickColorCircle = useCallback(() => {
+      setCardType(cardType)
+      onClicked()
+    }, [cardType, onClicked, setCardType])
+
+    return (
+      <Styled.CardTypeContainer>
+        <Styled.CardTypeColorCircle
+          cardType={cardType}
+          onClick={onClickColorCircle}
+        />
+        <Styled.CardTypeText>{cardType} 카드</Styled.CardTypeText>
+      </Styled.CardTypeContainer>
+    )
+  },
+  TouchableCardColorPickerAreEquals
+)
+
+const CardCreateModal = ({
+  isOpen,
+  setIsOpen,
+  setCardType,
+}: CardCreateModalProps) => {
+  const modalRef = useRef<HTMLDivElement>(null)
+  const cardTypes = Object.keys(
+    BackgroundColorAccordingToStartsWith
+  ) as CardType[]
+
+  const line1 = cardTypes.slice(0, 4)
+  const line2 = cardTypes.slice(4)
+
+  const onClickModalOutside: MouseEventHandler<HTMLDivElement> = useCallback(
+    (event) => {
+      if (event.target === modalRef.current) {
+        setIsOpen(false)
+      }
+    },
+    [modalRef, setIsOpen]
+  )
+
+  const onColorCircleClicked = useCallback(() => {
+    setIsOpen(false)
+  }, [setIsOpen])
+
+  return (
+    <Modal isOpen={isOpen} onClose={() => console.log('hi')}>
+      <Styled.Container ref={modalRef} onClick={onClickModalOutside}>
+        <Styled.ModalContainer>
+          <Styled.ContentContainer>
+            <Styled.ContentLineContainer>
+              {line1.map((cardType) => (
+                <TouchableCardColorPicker
+                  key={cardType}
+                  cardType={cardType}
+                  setCardType={setCardType}
+                  onClicked={onColorCircleClicked}
+                />
+              ))}
+            </Styled.ContentLineContainer>
+            <Styled.ContentLineSpace />
+            <Styled.ContentLineContainer>
+              {line2.map((cardType) => (
+                <TouchableCardColorPicker
+                  key={cardType}
+                  cardType={cardType}
+                  setCardType={setCardType}
+                  onClicked={onColorCircleClicked}
+                />
+              ))}
```

Comment:
```suggestion
              {cardTypes.map((cardType) => (
                <TouchableCardColorPicker
                  key={cardType}
                  cardType={cardType}
                  setCardType={setCardType}
                  onClicked={onColorCircleClicked}
                />
              ))}
```
굳이 slice하지 않고 위의 css로도 해결할 수 있을듯 합니다.